### DISCUSSION TITLE
I can't get/find a working copy of newer Cobbler for Ubuntu 20.04/22.04 LTS

### DISCUSSION THREAD

**Discussion by hustluh** [cobbler/cobbler#3492] [Q&A]
As the title states, I'm struggling to find a working method to get Cobbler on an Ubuntu 20.04 instance.. I tried following the install guide, but I guess I'm just missing something... or I'm newer than I thought?

Basically we were trying to use Foreman, and it's been a giant PITA to configure.. so I want to move us to Cobbler.. just looking for PXE and DHCP management for now.

**Comment by SchoolGuy** [comment]
So in general Cobbler should work on Ubuntu. Are there any specific errors? When Cobbler 3.3.3 was released we both tested Debian 10 & 11. Ubuntu shouldn't be that different in the Cobbler context.

**Reply by hustluh** [reply]
So, since Ubuntu 20.04 is based on Bullseye (Deb 11), should I run the processes for Deb 11 on the Ubuntu 20.04 box?

**Reply by hustluh** [reply]
If so... running into an error when I submit `make debs` after the a2enmod bits.

root:~/cobbler# make debs
creating: AUTHORS
/bin/bash: line 1: debuild: command not found
make: *** [Makefile:160: debs] Error 127

**Reply by hustluh** [reply]
and now this after fixing the debuild issue:

Cheetah Macros file note found. Using empty template.
/usr/lib/python3/dist-packages/Cheetah/Compiler.py:1686: UserWarning: You supplied an empty string for the source!
  warnings.warn("You supplied an empty string for the source!", )
tests.utils (unittest.loader._FailedTest) ... ERROR

======================================================================
ERROR: tests.utils (unittest.loader._FailedTest)
----------------------------------------------------------------------
ImportError: Failed to import test module: tests.utils
Traceback (most recent call last):
  File "/usr/lib/python3.8/unittest/loader.py", line 470, in _find_test_path
    package = self._get_module_from_name(name)
  File "/usr/lib/python3.8/unittest/loader.py", line 377, in _get_module_from_name
    __import__(name)
  File "/root/cobbler/.pybuild/cpython3_3.8_cobbler/build/tests/utils/__init__.py", line 6, in <module>
    import pytest
ModuleNotFoundError: No module named 'pytest'


----------------------------------------------------------------------
Ran 1 test in 0.000s

FAILED (errors=1)
E: pybuild pybuild:341: test: plugin distutils failed with: exit code=1: cd /root/cobbler/.pybuild/cpython3_3.8_cobbler/build; python3.8 -m unittest discover -v
dh_auto_test: error: pybuild --test -i python{version} -p 3.8 returned exit code 13
make: *** [debian/rules:15: build] Error 25
dpkg-buildpackage: error: debian/rules build subprocess returned exit status 2
debuild: fatal error at line 1182:
dpkg-buildpackage -us -uc -ui failed
make: *** [Makefile:160: debs] Error 29

**Reply by hustluh** [reply]
Forgot to include, the line before the "Cheetah Macros file note found" bit is: `I: pybuild base:217: cd /root/cobbler/.pybuild/cpython3_3.8_cobbler/build; python3.8 -m unittest discover -v`

**Reply by SchoolGuy** [reply]
To build a DEB the only thing needed on the host is docker. After that execute:

```
./docker/debs/build-and-install-debs.sh deb11 docker/debs/Debian_11/Debian11.dockerfile
```

It seems our documentation wasn't able to keep up. I will try to improve that for 3.3.4 and 3.4.0.
